### PR TITLE
Test Calcite doing query across single and sharding tables.

### PR DESCRIPTION
**Pull request by lmhmhl**
Ref #9117

### CODE REVIEW

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/resources/sql/calcite_data_1.sql [hunk 1]
```diff
@@ -16,8 +16,15 @@
  */
 
 DELETE FROM t_order_item_calcite;
+DELETE FROM t_order_item_calcite_sharding_0;
+ DELETE FROM t_order_item_calcite_sharding_1;
```
Comment (tristaZero) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/resources/sql/calcite_data_1.sql:20]: Please remove the first blank space.

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/resources/sql/jdbc_init_calcite_1.sql [hunk 2]
```diff
@@ -16,3 +16,5 @@
  */
 
 CREATE TABLE IF NOT EXISTS t_order_item_calcite (item_id INT NOT NULL, order_id INT NOT NULL, user_id INT NOT NULL, status VARCHAR(45) NULL, PRIMARY KEY (item_id));
+CREATE TABLE IF NOT EXISTS t_order_item_calcite_sharding_0 (item_id INT NOT NULL, order_id INT NOT NULL, user_id INT NOT NULL, status VARCHAR(45) NULL, remarks VARCHAR(45) NULL, PRIMARY KEY (item_id));
+ CREATE TABLE IF NOT EXISTS t_order_item_calcite_sharding_1 (item_id INT NOT NULL, order_id INT NOT NULL, user_id INT NOT NULL, status VARCHAR(45) NULL, remarks VARCHAR(45) NULL, PRIMARY KEY (item_id));
```
Comment (tristaZero) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/resources/sql/jdbc_init_calcite_1.sql:20]: Please remove the first blank space.

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi @lmhmhl 
Nice job! There only some check-style bits to improve.

**Comment by lmhmhl** [comment]
> Hi @lmhmhl
> Nice job! There are only some check-style bits to improve.

Thx.

**Review by tristaZero — APPROVED** [review]

**Comment by tristaZero** [comment]
Hi @lmhmhl ,

I wonder whether you are interested in #9152? This is one of `SQL federation` test series but a little difficult.
Take it easy and just follow your schedule.

**Comment by lmhmhl** [comment]
> Hi @lmhmhl ,
> 
> I wonder whether you are interested in #9152? This is one of `SQL federation` test series but a little difficult.
> Take it easy and just follow your schedule. Please comment that issue if you want.

Absolutely, I'd like to do that.

**Pull request merged**